### PR TITLE
Update options documentation

### PR DESCRIPTION
**Pull request by thejohnfreeman**

### CODE REVIEW

**Review by Bronek — APPROVED** [review]

File: BUILD.md [hunk 1]
```diff
@@ -376,11 +376,12 @@ stored inside the build directory, as either of:
...
 | `assert` | OFF | Enable assertions.
-| `reporting` | OFF | Build the reporting mode feature. |
 | `coverage` | OFF | Prepare the coverage report. |
-| `tests` | ON | Build tests. |
-| `unity` | ON | Configure a unity build. |
+| `reporting` | OFF | Build the reporting mode feature. |
 | `san` | N/A | Enable a sanitizer with Clang. Choices are `thread` and `address`. |
+| `tests` | OFF | Build tests. |
+| `unity` | ON | Configure a unity build. |
+| `xrpld` | OFF | Build the application, xrpld, and not just the library, libxrpl. |
```
Comment (intelliot) [inline BUILD.md:384]:
```suggestion
| `xrpld` | OFF | Build the xrpld (`rippled`) application, and not just the libxrpl library. |
```

**Review by seelabs — APPROVED** [review]
:+1:

File: BUILD.md [hunk 2]
```diff
@@ -376,11 +376,12 @@ stored inside the build directory, as either of:
 | Option | Default Value | Description |
 | --- | ---| ---|
 | `assert` | OFF | Enable assertions.
-| `reporting` | OFF | Build the reporting mode feature. |
 | `coverage` | OFF | Prepare the coverage report. |
-| `tests` | ON | Build tests. |
-| `unity` | ON | Configure a unity build. |
+| `reporting` | OFF | Build the reporting mode feature. |
```
Comment (godexsoft) [inline BUILD.md:380]: I assume this is later removed in the PR that removes reporting mode?

**Review by godexsoft — APPROVED** [review]

File: BUILD.md (hunk 2 above)
Comment (thejohnfreeman) [reply BUILD.md:380]: Yes, though it will need to be merged with this change after it hits `develop`.

**Pull request merged**